### PR TITLE
[MO] -> 1st part of removing dependency of logs

### PR DESCRIPTION
**Pull request by see-quick**
Signed-off-by: morsak <xorsak02@stud.fit.vutbr.cz>

### Type of change

- Refactoring

### Description

This PR is first part removing dependency to CO logs, specifically on reconciliation. 

### Checklist

- [ ] Make sure all tests pass

### CODE REVIEW

File: systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaConnectorUtils.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    private static final Logger LOGGER = LogManager.getLogger(KafkaConnectorUtils.class);
+
+    private KafkaConnectorUtils() {}
+
+    /**
+     * WaitForStabilityConnector method, verifying stability of connector
+     * @param connectorName connector name
+     * @param connectPodName connects2i or connect pod name
+     */
+    public static void waitForStabilityConnector(String connectorName, String connectPodName) {
```
Comment (tombentley) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaConnectorUtils.java]: Surely `waitForConnectorStability()` is a better name?

File: systemtest/src/main/java/io/strimzi/systemtest/Constants.java [hunk 2]
```diff
@@ -59,6 +59,9 @@
     long CO_OPERATION_TIMEOUT_POLL = Duration.ofSeconds(2).toMillis();
     long RECONCILIATION_INTERVAL = Duration.ofSeconds(30).toMillis();
 
+    int GLOBAL_STABILITY_COUNT = 20;
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/Constants.java]: Might be good to have some comments explaining what these are.

Comment (see-quick) [reply systemtest/src/main/java/io/strimzi/systemtest/Constants.java]: I would go for `GLOBAL_RECONCILIATION_COUNT = Duration.ofSeconds(30).toMillis() / GLOBAL_POLL_INTERVAL.` I had also idea to overload method `public static long waitFor(String description, long pollIntervalMs, long timeoutMs, BooleanSupplier ready)` to just to `waitFor(String description, BooleanSupplier ready)` and these two parameters set to some fix value inside parameter `long pollIntervalMs, long timeoutMs`

Comment (see-quick) [reply systemtest/src/main/java/io/strimzi/systemtest/Constants.java]: updated )

File: systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaConnectorUtils.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    public static void waitForConnectorStability(String connectorName, String connectPodName) {
+        // alternative to sync hassling AtomicInteger one could use an integer array instead
+        // not need to be final because reference to the array does not get another array assigned
+        int[] i = {0};
+
+        TestUtils.waitFor("Waiting for stability of connector " + connectorName, Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
+            () -> {
+                String availableConnectors = KafkaConnectUtils.getCreatedConnectors(connectPodName);
+                if (availableConnectors.contains(connectorName)) {
+                    LOGGER.info("Connector with name {} is present remaining seconds for stability {}", connectorName,
```
Comment (Frawless) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaConnectorUtils.java]: `"Connector with name {} is present. Remaining seconds for stability {}"`  ?

File: systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/controllers/StatefulSetUtils.java [hunk 4]
```diff
@@ -193,4 +193,22 @@ public static void waitForKafkaStatefulSetLabelsDeletion(String statefulSet, Str
...
+
+    public static void waitForNoRollingUpdate(String statefulSetName, Map<String, String> pods) {
+        // alternative to sync hassling AtomicInteger one could use an integer array instead
+        // not need to be final because reference to the array does not get another array assigned
+        int[] i = {0};
+
+        TestUtils.waitFor("Waiting for stability of rolling update will be not triggered", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
+            () -> {
+                if (!StatefulSetUtils.ssHasRolled(statefulSetName, pods)) {
+                    LOGGER.info("{} pods not rolling waiting, remaining seconds for stability {}", pods.toString(),
```
Comment (Frawless) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/kubeUtils/controllers/StatefulSetUtils.java]: Maybe we should use better log text like :`{} pods didn't roll. Remaining seconds for stability: {}`  wdyt?

**Pull request merged**